### DISCUSSION TITLE
Node.js/npm installer for Just

### DISCUSSION THREAD

**Discussion by brombal** [casey/just#1238] [Show and tell]
Hello,

I created this thing:

https://npmjs.com/package/just-install
https://github.com/brombal/just-install

It seemed like it would be a useful tool for those of us who work primarily in Node.js/npm, especially on development teams (as I do) where the simplicity of project development setup is a worthwhile goal.

It's an incredibly simple script that fetches the install bash script (https://just.systems/install.sh), executes it, and stores the resulting binary in `node_modules` where npm expects it to be. This makes the binary available within the project using `npx` or within a package.json script. 

While it does somewhat defeat the "globalness" feature that I recognize is part of Just's beauty, it has the benefit of not needing a separate install command (which until now I always have to go lookup on Just's website).

I am open to feedback, including admonishment for possibly re-inventing something that is already out there (that I couldn't find) or somehow "missing the point" of Just. But if none of that is the case, I hope this is useful for other fans of Just out there.

**Comment by casey** [comment]
Nice! I'm not sure I understand the use-case. Would it also be possible to use the install script, but install `just` to `node_modules/.bin`, like so:

```
curl --proto '=https' --tlsv1.2 -sSf https://just.systems/install.sh | bash -s -- --to node_modules/.bin
```

This requires remembering the long, complicated install command, so that's a downside.

**Reply by brombal** [reply]
I suppose you could do that, but in Node you aren't supposed to manually modify the contents of node_modules, as the npm package manager handles that for you (and would overwrite it on subsequent installs).

The use case is really just following the Node.js "purist" mindset of `npm install` being the only thing you should need to run to set up your project for development. Whether that's valid or not is debatable, but it makes it easier for junior devs on a team to get started.

**Reply by casey** [reply]
Gotcha, that makes sense. If you want to add a section to the readme, pointing node developers to `just-install`, that would be sweet. It would be good to outline the reasoning, i.e. you could use the installer, but if you want `npm install` to the the only thing that people have to run, you can use `just-install.

**Reply by brombal** [reply]
Thanks @casey, done! https://github.com/brombal/just/pull/1

**Reply by agilgur5** [reply]
> Thanks @casey, done! [brombal#1](https://github.com/brombal/just/pull/1)

For reference for other readers, this was moved to https://github.com/casey/just/pull/1241 per https://github.com/brombal/just/pull/1#issuecomment-1160705045

**Comment by agilgur5** [comment]
For reference for other readers searching for NPM or `package.json`, another variant was recently added via [`rust-just`](https://github.com/gnpaone/rust-just) in https://github.com/casey/just/pull/2361#issuecomment-2365008932 / #2384

**Comment by goyalyashpal** [comment]
[nix-download]: https://nixos.org/download/#nix-install-windows "Download | Nix"
[nix-features]: https://nixos.org/explore/ "Explore | Nix"

* i see the reasoning behind single command to setup the project on new system, 
* but it seem _absurd_ to mingle executables in language specific environment with which they have no direct business whatsoever


that's why i am super glad to have found the [nix pkg manager][nix-download] which is a [language agnostic manager][nix-features]  for development environments and projects.  

it makes all such tasks _trivial_ to carry out, removing the hodge-podge.  and gives so much peace of mind to me as a developer. 


----
well, it's kinda weird to me that similar concerns are shared by the author of this discussion over at https://github.com/casey/just/pull/2361#issuecomment-2466416027 and following, but ...
